File: PostBox/src/main/java/com/mvp/presenter/BoxPresenter.java
package com.mvp.presenter;

import com.mvp.model.BoxModel;
import com.mvp.model.IBoxModel;
import com.mvp.view.IBoxView;

public class BoxPresenter {
    private IBoxView boxView;
    private IBoxModel boxModel;

    public BoxPresenter(IBoxView boxView) {
        this.boxView = boxView;
        this.boxModel = new BoxModel();
    }

    public void findSuitableBoxFromModel(float length, float width, float height) {
        System.out.println("Presenter : find SuitableBox From Model");
        boxModel.setBoxSize(length, width, height);
        returnCheckResult();
    }

    private void returnCheckResult() {
        String boxResult = boxModel.throwVerifiedBoxBackToClient();
        if (boxResult != null && boxResult.length() > 0) {
            // update view
            boxView.reportResultToClient(boxResult);
        } else {
            boxView.reportResultToClient("no suitable box for you sorry ...");
        }
    }
}

File: PostBox/src/main/kotlin/com/mvp/presenter/BoxPresenter.kt
package com.mvp.presenter.kt

import com.mvp.model.kt.BoxModel
import com.mvp.model.kt.IBoxModel
import com.mvp.view.kt.IBoxView

class BoxPresenter(private val boxView: IBoxView,
                   private val boxModel: IBoxModel = BoxModel()) {

    fun findSuitableBoxFromModel(length: Float?, width: Float?, height: Float?) {
        println("Presenter : try to find suitableBox from model")
        boxModel.setBoxSize(length, width, height)
        returnCheckResult()
    }

    private fun returnCheckResult() {
        val result = boxModel.returnVerifiedBoxToClient()
        if (result != null && result.isNotEmpty()) {
            boxView.reportResultToClient(result)
        } else {
            boxView.reportResultToClient("no suitable box for you sorry ...")
        }
    }

}